Introduce download-sbom docs (#457)
buildpacks/docs: content/docs/features/bill-of-materials.md

+++
title="Structured Bill of Materials"
summary="A Software `Structured Bill-of-Materials` (`SBoM`) gives you a layer-by-layer view of what's inside your container in a variety of formats."
+++

## Summary

> A Software **Structured-Bill-of-Materials** (`SBoM`) provides information necessary to know what's inside your container and how it was constructed.

Reviewer: ```suggestion
A **Structured-Bill-of-Materials** (`SBoM`) provides information necessary to know what's inside your container and how it was constructed.
```

## Viewing Bill of Materials

> You can use the `download-sbom` command to inspect your app for it's `Structured-Bill-of-Materials`. The following command will download the application layer containing the `SBoM` files to `./layers/sbom/...`.

Reviewer: ```suggestion
You can use the `download-sbom` command to inspect your app for its `Structured-Bill-of-Materials`. The following command will download the application layer containing the `SBoM` files to `./layers/sbom/...` on your local filesystem.
```

> You can use the `download-sbom` command to inspect your app for its `Structured-Bill-of-Materials`. The following command will download the application layer containing the `SBoM` files to `./layers/sbom/...` on your local filesystem.

Reviewer: ```suggestion
You can use the `sbom download` command to inspect your app for its `Structured-Bill-of-Materials`. The following command will download the application layer containing the `SBoM` files to `./layers/sbom/...` on your local filesystem.
```

```bash

> pack download-sbom your-image-name

Reviewer: ```suggestion
pack sbom download your-image-name
```

```

You can also choose to download the `SBoM` from an image hosted in a remote registry, as opposed to an image hosted in a Docker daemon. You use the `--remote` flag to do so.

```bash

> pack download-sbom your-image-name --remote

Reviewer: ```suggestion
pack sbom download your-image-name --remote
```

```

> The following is a sample directory structure for an `SBoM` layer downloaded to the local filesystem. The files are named in the following pattern `sbom.<FORMAT>.json`, where FORMAT can be one of the Structured-Bill-of-Material formats allowed by Cloud Native Buildpacks tooling: `cdx`, `spdx`, or `syft`. Buildpack authors may choose to generate `SBoM` for the entire buildpack, or individual layers designated by the buildpack.

Reviewer: The current phrasing is technically accurate, but it could lead to the reader thinking of lots of edge-cases.  For example, what happens if one layer contains `syft` SBoM and another layer contains `cdx`.  I think we can simplify the paragraph and focus on your syft example:

```suggestion
Cloud Native Buildpacks support `SBoM` metadata in [CycloneDX](https://cyclonedx.org/), [Syft](https://github.com/anchore/syft) or [Spdx](https://spdx.dev/) formats.  The following example demonstrates `syft` format `SBoM` metadata to the local filesystem.  The combined metadata from all of the `sbom.syft.json` files is the image `SBoM`.   Where CycloneDX `SBoM` metadata is generated, the files are named `sbom.cdx.json`.  Similarly, Spdx files are named `sbom.cdx.json`.
```
